testing/citest/tests/kube_smoke_test.py

```diff
@@ -390,47 +395,382 @@ def make_deploy_stage(self, imageSource=None, requisiteStages=[], **kwargs):
+                  'containers': [
+                      {
+                          'name': 'Statefulset-Pipeline',
+                          'imageDescription': {
+                              'repository': 'nginx'
+                          },
+                          'imagePullPolicy': 'ALWAYS',
+                          'limits': {
+                              'memory': '200Mi'
+                          },
+                          'name': 'nginx',
+                          'requests': {
+                              'cpu': '100m',
+                              'memory': '200Mi'
+                          }}
+                  ],
+                  'kind': 'StatefulSet',
+                  'volumeSources': [],
+                  'provider': 'kubernetes',
+                  'regions': [self.TEST_NAMESPACE],
+                  'region': self.TEST_NAMESPACE
+              }
+          ]
+      }
+      result.update(kwargs)
+      return result
+
+  def save_delete_statefulset_pipeline(self):
+      deploy_stage = self.make_delete_stage_statefulset()
+      job = dict(
+          appConfig={},
+          keepWaitingPipelines='false',
+          application=self.TEST_APP,
+          name='statefulset-delete-pipeline',
+          lastModifiedBy='anonymous',
+          limitConcurrent='true',
+          parallel='true',
+          stages=[deploy_stage]
+      )
+      payload = self.agent.make_json_payload_from_kwargs(**job)
+      expect_match = {key: jp.EQUIVALENT(value)
+                      for key, value in job.items()}
+      expect_match['stages'] = jp.LIST_MATCHES(
+          [
+              jp.DICT_MATCHES({key: jp.EQUIVALENT(value)
+                               for key, value in deploy_stage.items()})])
+
+      builder = st.HttpContractBuilder(self.agent)
+      (builder.new_clause_builder('Has Pipeline',
+                                  retryable_for_secs=15)
+       .get_url_path(
+          'applications/{app}/pipelineConfigs'.format(app=self.TEST_APP))
+       .contains_match(expect_match))
+      return st.OperationContract(
+          self.new_post_operation(title='create_delete statefulset',
+                                  data=payload, path='pipelines',
+                                  status_class=st.SynchronousHttpOperationStatus),
+          contract=builder.build())
+
+  def save_delete_daemonset_pipeline(self):
+      deploy_stage = self.make_delete_stage_daemonset()
+      job = dict(
+          appConfig={},
+          keepWaitingPipelines='false',
+          application=self.TEST_APP,
+          name='daemonset-delete-pipeline',
+          lastModifiedBy='anonymous',
+          limitConcurrent='true',
+          parallel='true',
+          stages=[deploy_stage]
+      )
+      payload = self.agent.make_json_payload_from_kwargs(**job)
+      expect_match = {key: jp.EQUIVALENT(value)
+                      for key, value in job.items()}
+      expect_match['stages'] = jp.LIST_MATCHES(
+          [
```

Comment:
Formatting looks off

Reply:
No Its not if you see existing test case 'create_find_image_pipeline', that also has the same